sdk/python/lib/pulumi/dynamic/__main__.py

```diff
@@ -0,0 +1,176 @@
+        return proto.UpdateResponse(**fields)
+
+    def Delete(self, request, context):
+        id_ = request.id
+        props = rpc.deserialize_properties(request.properties)
+        provider = get_provider(props)
+        provider.delete(id_, props)
+        return empty_pb2.Empty()
+
+    def Cancel(self, request, context):
+        return empty_pb2.Empty()
+
+    def Create(self, request, context):
+        props = rpc.deserialize_properties(request.properties)
+        provider = get_provider(props)
+        result = provider.create(props)
+        outs = result.outs
+        outs[PROVIDER_KEY] = props[PROVIDER_KEY]
+
+        loop = asyncio.new_event_loop()
+        outs_proto = loop.run_until_complete(rpc.serialize_properties(outs, {}))
+        loop.close()
+
+        fields = {"id": result.id, "properties": outs_proto}
+        return proto.CreateResponse(**fields)
+
+    def Check(self, request, context):
+        olds = rpc.deserialize_properties(request.olds)
+        news = rpc.deserialize_properties(request.news)
+        if news[PROVIDER_KEY] == rpc.UNKNOWN:
+            provider = get_provider(olds)
+        else:
+            provider = get_provider(news)
+
+        result = provider.check(olds, news)
+        inputs = result.inputs
+        failures = result.failures
+
+        inputs[PROVIDER_KEY] = news[PROVIDER_KEY]
+
+        loop = asyncio.new_event_loop()
+        inputs_proto = loop.run_until_complete(rpc.serialize_properties(inputs, {}))
+        loop.close()
+
+        failures_proto = [proto.CheckFailure(f.property, f.reason) for f in failures]
+
+        fields = {"inputs": inputs_proto, "failures": failures_proto}
+        return proto.CheckResponse(**fields)
+
+    def Configure(self, request, context):
+        fields = {"acceptSecrets": False}
+        return proto.ConfigureResponse(**fields)
+
+    def GetPluginInfo(self, request, context):
+        fields = {"version": "0.1.0"}
+        return proto.PluginInfo(**fields)
+
+    def Read(self, request, context):
+        id_ = request.id
+        props = rpc.deserialize_properties(request.properties)
+        provider = get_provider(props)
+        result = provider.read(id_, props)
+        outs = result.outs
+        outs[PROVIDER_KEY] = props[PROVIDER_KEY]
+
+        loop = asyncio.new_event_loop()
+        outs_proto = loop.run_until_complete(rpc.serialize_properties(outs, {}))
+        loop.close()
+
+        fields = {"id": result.id, "properties": outs_proto}
+        return proto.ReadResponse(**fields)
+
+    def __init__(self):
+        pass
+
+def main():
+    monitor = DynamicResourceProviderServicer()
+    server = grpc.server(futures.ThreadPoolExecutor(max_workers=4))
+    provider_pb2_grpc.add_ResourceProviderServicer_to_server(monitor, server)
+    port = server.add_insecure_port(address="0.0.0.0:0")
+    server.start()
+    print(port)
+    try:
+        while True:
+            time.sleep(_ONE_DAY_IN_SECONDS)
```

Comment:
what's the idea here?

Reply:
This followed what I saw as recommended practice for Python gRPC servers - e.g.: https://github.com/grpc/grpc/blob/v1.22.0/examples/python/route_guide/route_guide_server.py

Honestly not sure why they choose to do it this specific day, but I suppose a long sleep in a loop is as good as anything for this purpose.